### PR TITLE
[nfr] setup basic login features

### PR DESCRIPTION
**Pull request by TaylorJCrawford**

### CODE REVIEW

File: controller/employeeController.ts [hunk 1]
```diff
@@ -13,6 +13,8 @@ module.exports = function (app: Application) {
             console.error(e);
         }
 
-        res.render('list-delivery-employees', { deliveryEmployees: data })
+        let tokenValue: String = req.session.token
+
```
Comment (glenn-hamilton) [inline controller/employeeController.ts]: You can clean this up with 

```
const { token }: String = req.session;
res.render('list-delivery-employees', { deliveryEmployees: data, token })
```

^ Completely untested

File: views/login.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        {% endif %}
+    </div>
+    <form action="/login" method="post">
+        <div class="form-group">
+            <label for="username">Username</label>
+            <input type="text" id="username" name="username" class="form-control" value="{{ username }}" required>
+        </div>
+        <div class="form-group">
+            <label for="password">Password</label>
+            <input type="password" id="password" name="password" class="form-control" value="{{ password }}" required>
```
Comment (glenn-hamilton) [inline views/login.html]: You wouldn't want to return the password here

File: views/templates/master.html [hunk 3]
```diff
@@ -18,20 +18,28 @@
...
+                {% endif %}
               </ul>
               <ul class="navbar-nav">
+                {% if token == undefined %}
                   <li class="nav-item">
                       <a class="nav-link" href="login" id="nav-cust-link">Login</a>
                   </li>
                   <li class="nav-item">
-                      <a class="nav-link"href="new_user" id="nav-cust-link">Signup</a>
+                      <a class="nav-link btn disabled"href="/" id="nav-cust-link">Signup</a>
```
Comment (glenn-hamilton) [inline views/templates/master.html]: Should this go to the root?

File: controller/employeeController.ts (hunk 1 above)
Comment (TaylorJCrawford) [reply controller/employeeController.ts]: Done

File: views/login.html (hunk 2 above)
Comment (TaylorJCrawford) [reply views/login.html]: Done

File: views/templates/master.html (hunk 3 above)
Comment (TaylorJCrawford) [reply views/templates/master.html]: Done

File: controller/employeeController.ts [hunk 4]
```diff
@@ -13,6 +13,7 @@ module.exports = function (app: Application) {
             console.error(e);
         }
 
-        res.render('list-delivery-employees', { deliveryEmployees: data })
+        const { token } = req.session;
```
Comment (shaunganley) [inline controller/employeeController.ts:16]: any reason for this?

File: services/authService.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Login } from "../model/auth";
+
+const axios = require('axios');
+
+module.exports.login =async function (login: Login): Promise<void> {
```
Comment (shaunganley) [inline services/authService.ts]: aren't you supposed to return the token?

**Review by shaunganley — CHANGES_REQUESTED** [review]

File: controller/employeeController.ts (hunk 4 above)
Comment (TaylorJCrawford) [reply controller/employeeController.ts:16]: We are checking the token value to hide links based on the value of token in templates/master

File: services/authService.ts (hunk 5 above)
Comment (TaylorJCrawford) [reply services/authService.ts]: All fixed, thank you!

**Review by shaunganley — APPROVED** [review]

**Pull request merged**